internal/dag/gatewayapi_processor.go

```diff
@@ -1402,10 +1402,22 @@ func gatewayGRPCHeaderMatchConditions(matches []gatewayapi_v1alpha2.GRPCHeaderMa
 	seenNames := sets.New[string]()
 
 	for _, match := range matches {
-		// "Exact" is the default if not defined in the object, and
-		// the only supported match type.
-		if match.Type != nil && *match.Type != gatewayapi_v1beta1.HeaderMatchExact {
-			return nil, fmt.Errorf("GRPCRoute.Spec.Rules.Matches.Headers: Only Exact match type is supported")
+		// "Exact" and "RegularExpression" are the only supported match types. If match type is not specified, use "Exact" as default.
+		var matchType string
+		if match.Type == nil {
+			matchType = HeaderMatchTypeExact
+		} else {
+			switch *match.Type {
```

Comment:
Could simplify this a bit using a generic helper we have for defaulting a nil pointer:
```suggestion
		switch ref.Val(match.Type, gatewayapi_v1beta1.HeaderMatchExact) {
```